core/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java

```diff
@@ -225,32 +226,41 @@ void save(Path configDir) throws Exception {
         }
     }
 
-    /** Returns the names of all settings in this keystore. */
     public Set<String> getSettings() {
         return settingNames;
     }
 
+    @Override
+    public boolean hasSetting(String setting) {
+        return settingNames.contains(setting);
+    }
+
     // TODO: make settings accessible only to code that registered the setting
     /** Retrieve a string setting. The {@link SecureString} should be closed once it is used. */
-    SecureString getStringSetting(String setting) throws GeneralSecurityException {
-        KeyStore.Entry entry = keystore.get().getEntry(setting, keystorePassword.get());
-        if (entry instanceof KeyStore.SecretKeyEntry == false) {
-            throw new IllegalStateException("Secret setting " + setting + " is not a string");
+    @Override
+    public SecureString getString(String setting) {
```

Comment:
Sure, we could throw it. I was thinking ahead to Vault integration, but I guess that could make GeneralSecurityExceptions?